### ISSUE TITLE
plug won't connect to wi-fi network after update 

### ISSUE THREAD

**Issue by estevaozr** [KaufHA/PLF10#19]
Hello,

I am having an issue after trying to update the firmware of the plug to the latest version of the repository. Now the plug is not connecting the the wi-fi network and is does not appear to respond to the button to turn it on/off. It also appears to not be responding to me holding the button for 30s for it to forget the wifi credentials as I can't see any AP from it (checked with WIFI analyzer on android phone). For reference I have used the plug for several months with no apparent issue, so the hardware *should* be mostly OK.

The updated `kauf-plug.yaml` is what is the current latest commit on the repo (commit `2973da17141afefd2ccaa19c21021d209de596b0`) and I only changed the name and wifi connections as follows:

```
diff kauf-plug.yaml kauf-plug-f6e9b1___new.yaml 
12,14c12,14
<   name: kauf-plug              # **** CHANGE DEVICE NAME TO SOMETHING UNIQUE PER DEVICE.  RENAME YAML FILE TO SAME NAME.    ****
<                                # **** USE DASHES (-) INSTEAD OF SPACES OR UNDERSCORE (_).  USE ONLY LOWER CASE LETTERS.     ****
<   friendly_name: Kauf Plug     # **** CHANGE FRIENDLY NAME TO SOMETHING UNIQUE PER DEVICE                                   ****
---
>   name: kauf-plug-f6e9b1                # **** CHANGE DEVICE NAME TO SOMETHING UNIQUE PER DEVICE.  RENAME YAML FILE TO SAME NAME.    ****
>                                         # **** USE DASHES (-) INSTEAD OF SPACES OR UNDERSCORE (_).  USE ONLY LOWER CASE LETTERS.     ****
>   friendly_name: Kauf Plug (f6e9b1)     # **** CHANGE FRIENDLY NAME TO SOMETHING UNIQUE PER DEVICE                                   ****
160,161c160,161
<   ssid: initial_ap            # !secret wifi_ssid
<   password: asdfasdfasdfasdf  # !secret wifi_password
---
>   ssid:     !secret wifi_ssid
>   password: !secret wifi_password

```
I've built the updated firmware file using ESPHome v2023.6.3 (which I think is the latest, or at least a recent version) on a RPi4 (which I think should not matter) with no apparent issue.

After building the fw file and compressing it with tar.gz (down to 361378 bytes, so it should be OK for an OTA update) I performed the OTA update and the device page said it was successful. As I mentioned right now I have a plug that is not connecting to a wifi network, red LED is dimly on and is not responding the the button.

I'd like for some help to fix / un-brick the device so I could continue using it.

Let me know if I was unlucky or if I did something wrong, and I am looking forward for some help.

p.s.: I will attach the "backup" and the "new" versions of the yaml configurations for the device that I used for reference. Let me know if me uploading the actual FW binary would help as well.

The top plug is the one with the issue (it uses the `kauf-plug-f6e9b1___new.yaml` configuration), the bottom one is using a configuration very similar (except the unique names) to the old configuration that I've attached (the `kauf-plug-f6e9b1___backup.yaml` configuration)
[configurations.zip](https://github.com/KaufHA/PLF10/files/11991364/configurations.zip)

![photo_2023-07-08_10-21-05](https://github.com/KaufHA/PLF10/assets/59417497/8ff41b4b-d718-44a5-ae23-5ace44c0a131)

**Comment by bkaufx** [comment]
Probably something went wrong with the flash.  Did the same thing work on other plugs?  If you send me an email I'll send a replacement.  brian@kaufha.com

**Comment by estevaozr** [comment]
Hey @bkaufx, I didn't try flashing another plug after the first one I tried failed.

I recently saw a review of the product that mentioned this:

```
If you need to update your firmware for any reason, like esphome updating or needing to tweak any settings, you MUST flash their minimal firmware before flashing your real firmware because there is not enough space accept it ota (...)
```
I didn't have to do this when I first bought the product and when I did a first upgrade afterwards (my comment here was my 2nd upgrade of the configuration).

Do you think the procedure of flashing the `kauf-plug-minimal.yaml` before trying to flash the `kauf-plug.yaml` could help in future upgrades? If so maybe it would be a good idea to add this information on the repository documentation.

I will send an e-mail for a replacement unit shortly. Thanks for your attention

**Comment by JayGoldberg** [comment]
I also flashed based in mimimal config and the 30s button press for AP recovery (captive portal) does not appear to work.

Instead the device attempts to join (STA mode, not AP) a network called `initial_ap` as the YAML is configured, not the factory AP mode `'kauf_plug_XXX`  (as I recall). Can we update the repo to reflect the firmware shipped on the device?

**Comment by JayGoldberg** [comment]
Yes, I solved the device not getting into AP mode by changing to `ap : {}` which is an [ESPHome default AP config](url) with no password and a device name based AP name like `kauf-plug-XXXXXX` where XXXXXX is the last 6 of the MAC address.

So now it has the intended behavior for me of enabling AP mode after 60 sec, without having to clear the device with the 30 second input button down.

I'm leaving save_wifi_sta("initial_ap","asdfasdfasdfasdf") since this is a nice alternative recovery method if the AP mode and captive portal fail.
